File: Server/knight-online/shared/src/main/java/com/knightonline/shared/data/enums/PremiumEnum.java
package com.knightonline.shared.data.enums;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

/**
 * @author Mamaorha
 *
 */
public enum PremiumEnum implements Serializable
{
	NONE(0), BRONZE(1), SILVER(2), GOLD(3), PLATINUM(4);
	private static Map<Integer, PremiumEnum> namesMap = new HashMap<>();

	private Integer value;
	
	PremiumEnum(Integer value)
	{
		this.value = value;
	}

	public Integer getValue()
	{
		return this.value;
	}
	
	public static PremiumEnum forValue(Integer value)
	{
		return namesMap.get(value);
	}
	
	static
	{
		final PremiumEnum[] values = PremiumEnum.values();
		
		for (PremiumEnum temp : values)
		{
			namesMap.put(temp.getValue(), temp);
		}
	}
}

File: Client/WarFare/PremiumEnum.cpp
#include "PremiumEnum.h"

PremiumEnum & PremiumEnum::NONE = createEnum("NONE", 0);
PremiumEnum & PremiumEnum::BRONZE = createEnum("BRONZE", 1);
PremiumEnum & PremiumEnum::SILVER = createEnum("SILVER", 2);
PremiumEnum & PremiumEnum::GOLD = createEnum("GOLD", 3);
PremiumEnum & PremiumEnum::PLATINUM = createEnum("PLATINUM", 4);